cdap-storage-ext-spanner/src/main/java/io/cdap/cdap/storage/spanner/SpannerStructuredTableAdmin.java

```diff
@@ -108,6 +111,27 @@ public void createOrUpdate(StructuredTableSpecification spec)
     }
   }
 
+  @Override
+  public void updateOrDeferCreate(StructuredTableSpecification spec)
+      throws TableSchemaIncompatibleException, IOException {
+    if (exists(spec.getTableId())) {
+      tryUpdatingTable(spec);
```

Comment:
Here's the source of confusion. We should either do batch or synchronously call create/update.